### PR TITLE
[feature](insert)support external hive truncate table DDL

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

Issue Number: #31442

1. support hive3 'truncate table'
2. forbiden hive2 'truncate table' because of it is not supported by HMS Client API on hive2

### CODE REVIEW

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run external

**Comment by wsjz** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/hive/ThriftHMSCachedClient.java [hunk 1]
```diff
@@ -244,6 +244,23 @@ public void dropTable(String dbName, String tblName) {
...
+                ugiDoAs(() -> {
+                    client.client.truncateTable(dbName, tblName, partitions);
+                    return null;
+                });
+            } catch (Exception e) {
+                client.setThrowable(e);
+                throw e;
+            }
+        } catch (Exception e) {
+            throw new HMSClientException("failed to check if table %s in db %s exists", e, tblName, dbName);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/hive/ThriftHMSCachedClient.java:260]: change msg

File: fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergMetadataOps.java [hunk 2]
```diff
@@ -178,4 +179,8 @@ public void dropTable(DropTableStmt stmt) throws DdlException {
         db.setUnInitialized(true);
     }
 
+    @Override
+    public void truncateTable(String dbName, String tblName, List<String> partitions) {
+        throw new UnsupportedOperationException("Iceberg truncate table is not supported.");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/iceberg/IcebergMetadataOps.java:184]: Tra
```suggestion
        throw new UnsupportedOperationException("Truncate Iceberg table is not supported.");
```

**Comment by wsjz** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by wuwenchi — APPROVED** [review]

**Pull request merged**